### ISSUE TITLE
Refactor navigation

### ISSUE THREAD

**Issue by czbar** [czbar/ChessForge#537]
When we go from one view to another, e.g. changing a chapter following a reference, we may want to easily go back to where we navigated from.  Browser style arrows would be of help.

**Comment by czbar** [comment]
Design suggestion: create a class that uniquely identifies position by:
- Workbook GUID
- Chapter GUID
- ArticleType
- Article GUID
- Selected Line
- Selected Node

**Comment by eradecka** [comment]
Generally works fine however there is a redundant arrow at the end that takes you back to the page that was viewed as the second.

Example.


1. I opened Chess Forge and Chapter list appeared. I expanded Chapter 2 and highlighted Game 4:

![image](https://github.com/czbar/ChessForge/assets/110824781/776fdeee-fcc0-4b8f-9775-0f62bb90578c)

2. I double clicked the above Game 4 to open it and then moved to the previous game: Game 3

![image](https://github.com/czbar/ChessForge/assets/110824781/6e849c44-d968-45b2-abce-e2a0a6e88956)

3. Now I click Study and highlight move 15.Be7

![image](https://github.com/czbar/ChessForge/assets/110824781/4da81ffd-4936-4798-bf12-710ad02ae22b)

4. Now I clicked on Exercises which are empty and started to use navigation Back

Back 1 - works fine and shows Study with highlighted move 15.Be7
Back 2 - wors fine and shows Game 3
Back 3 - wors fine and shows Game 4
Back 4 - wors fine and shows Chapter List
Back 5 - this should be greyed out as I started a session from Chapter List above. When I click it it takes me to Study Move 15 Be7 which was the screen after Back 1
